casper/message.py

```diff
@@ -1,46 +1,45 @@
-"""The message module ... """
+"""The message module defines an abstract message class  """
+import random as r
 from casper.justification import Justification
 
+
 class Message(object):
     """Message/bet data structure for blockchain consensus"""
     def __eq__(self, message):
         if message is None:
             return False
-        return self.hash == message.hash
+        return self.message_hash == message.message_hash
 
     def __ne__(self, message):
         return not self.__eq__(message)
 
     def __init__(self, estimate, justification, sender):
-        assert isinstance(justification, Justification), "expected justification a Justification!"
+        assert isinstance(justification, Justification), "justification should be a Justification!"
 
-        # set the necessary variables
         self.sender = sender
         self.estimate = estimate
         self.justification = justification
 
-        # The sequence number makes certain operations more
-        # efficient (like checking if bets are later).
-        if self.sender not in self.justification.latest_messages:
-            self.sequence_number = 0
-        else:
+        if self.sender in self.justification.latest_messages:
             latest_message = self.justification.latest_messages[self.sender]
             self.sequence_number = latest_message.sequence_number + 1
+        else:
+            self.sequence_number = 0
 
         # The "display_height" of bets are used for visualization of views
         if not any(self.justification.latest_messages):
             self.display_height = 0
         else:
-            max_height = max(self.justification.latest_messages[validator].display_height \
-                            for validator in self.justification.latest_messages)
-
+            max_height = max(
+                self.justification.latest_messages[validator].display_height
+                for validator in self.justification.latest_messages
+            )
             self.display_height = max_height + 1
 
-        self.hash = self.__hash__()
+        self.message_hash = r.randint(0, 1000000)
```

Comment:
I think i'd prefer calling this `salt` or something, including it in the __hash__ method (like it is), then still using `message.hash` to check equality, etc.

For our current structure, i agree we are going to need something like this, but I just don't like the look of `message.message_hash`.